### PR TITLE
Release candidate 1.4.0

### PR DESCRIPTION
**Pull request by Aratz**
Release candidate for v1.4.0

Includes the following:

### `Added`

- [#148](https://github.com/nf-core/demultiplex/pull/148) Update CODEOWNERS to use GitHub teams

### `Changed`

- [#141](https://github.com/nf-core/demultiplex/pull/141) Updated template to nf-core/tools v2.10
- [#152](https://github.com/nf-core/demultiplex/pull/152) Updated bclconvert module to 4.2.4

### `Fixed`

- [#127](https://github.com/nf-core/demultiplex/pull/127) Add `singularity.registry = 'quay.io'` and bump NF version to 23.04.0
- [#140](https://github.com/nf-core/demultiplex/pull/140) Make it possible to skip MultiQC, fix error raising
- [#145](https://github.com/nf-core/demultiplex/pull/145) Fix MultiQC report generation
- [#152](https://github.com/nf-core/demultiplex/pull/152) Close [#150](https://github.com/nf-core/demultiplex/issues/150)
- [#157](https://github.com/nf-core/demultiplex/pull/157) Fix bcl2fastq and bclconvert publishDir
- [#158](https://github.com/nf-core/demultiplex/pull/158) Update all modules

## `Removed`

- [#130](https://github.com/nf-core/demultiplex/pull/130) Remove `public_aws_ecr` profile.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,93 +4,73 @@ on:
...
-          python-version: ${{ matrix.python-version }}
-
-      - name: Conda clean
-        if: matrix.profile == 'conda'
-        run: conda clean -a
-
+      # Run nf-test
       - name: Run nf-test
-        run: nf-test test --profile=test,${{ matrix.profile }} tests/${{ matrix.component }}/${{ matrix.tags }}/*.nf.test --junitxml=${{ matrix.tags }}.xml
+        run: nf-test test tests/pipeline/ --profile=test,${{ matrix.profile }} --junitxml=test.xml
```
Comment (maxulysse) [inline .github/workflows/ci.yml:63]: I feel like using tag would be more efficient, but good enough for me

File: assets/schema_input.json [hunk 2]
```diff
@@ -25,11 +25,11 @@
                 "minimum": 1,
                 "maximum": 8
             },
-            "run_dir": {
```
Comment (maxulysse) [inline assets/schema_input.json:28]: what happened to `run_dir`?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Emiller88) [reply .github/workflows/ci.yml:63]: It was written pre-tags 😉

File: assets/schema_input.json (hunk 2 above)
Comment (Emiller88) [reply assets/schema_input.json:28]: I don't think we're using `from_samlesheet` yet, so this won't matter until #102

**Review by Emiller88 — APPROVED** [review]
Awesome!

**Review by maxulysse — APPROVED** [review]

**Comment by maxulysse** [comment]
Thanks a lot @Aratz <3

**Pull request merged**